src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -277,6 +290,45 @@ private IList<ContentItem> FindItemsImplementation(PatternSet definition, IEnume
             return itemsList;
         }
 
+        private string GetRelatedFileExtensionProperty(string assemblyPath, IEnumerable<Asset> assets)
+        {
+            //E.g. if path is "lib/net472/A.B.C.dll", the prefix will be "lib/net472/A.B.C."
+            string assemblyPrefix = assemblyPath.Substring(0, assemblyPath.LastIndexOf('.') + 1);
```

Comment:
What if `assemblyPath` has the value of `"a.b."`, will this line work?

Reply:
If assembly path doesn't have any `.` in it then `assemblyPrefix` end up `string.Empy`.

```
string assemblyPath = "C:\\winexe";
string assemblyPrefix = assemblyPath.Substring(0, assemblyPath.LastIndexOf('.') + 1);
Console.WriteLine(assemblyPrefix == string.Empty);
```

Reply:
`GetRelatedFileExtensionProperty` will only be applied when `contentItem.Properties.ContainsKey("assembly")`.
And according to https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/ContentModel/ManagedCodeConventions.cs#L26-L28, only ".dll", ".winmd", ".exe" will be recognized as assembly property (and _._).